src/main/java/netcracker/study/controller/Bootstrap.java

```diff
@@ -0,0 +1,84 @@
+package netcracker.study.controller;
+
+import lombok.Getter;
+import lombok.Setter;
+import netcracker.study.api.controller.IBootstrap;
+import netcracker.study.api.repository.IProjectRepository;
+import netcracker.study.api.repository.ITaskRepository;
+import netcracker.study.api.service.IProjectService;
+import netcracker.study.api.service.ITaskService;
+import netcracker.study.command.AbstractCommand;
+import netcracker.study.error.NoSuchCommandsException;
+import netcracker.study.repository.ProjectRepository;
+import netcracker.study.repository.TaskRepository;
+import netcracker.study.service.ProjectService;
+import netcracker.study.service.TaskService;
+import org.reflections.Reflections;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.Set;
+
+@Getter
+@Setter
+public class Bootstrap implements IBootstrap {
+
+    private final IProjectRepository projectRepository = new ProjectRepository();
+
+    private final ITaskRepository taskRepository = new TaskRepository();
+
+    private final IProjectService projectService = new ProjectService(projectRepository);
+
+    private final ITaskService taskService = new TaskService(taskRepository);
+
+    private final Scanner scanner = new Scanner(System.in);
+
+    private final Reflections reflections = new Reflections("netcracker.study.command");
+
+    private final Set<Class<? extends AbstractCommand>> commandClasses = reflections.getSubTypesOf(AbstractCommand.class);
```

Comment:
Use factory pattern. Try to avoid using third party (especially this one looks like works with reflection) where its not needed and same logic can be written in pure Java easily

Reply:
But I create one instance of each command. Do I need a factory here? I used reflection because when creating a new command class I didn't have to add it anywhere else